### ISSUE TITLE
Permission model

### ISSUE THREAD

**Issue by marcoscaceres** [w3c/webpayments-payment-apps-api#363]
We might need to revisit the permission model for this spec. In particular, I'm a little bit concerned that any site registering a service worker can grab an instance of PaymentManger and start adding instruments. 

This is particularly concerning for things like "basic-card", where any arbitrary site can be like, "I can do basic card for you" and insert a random instrument (e.g., just to annoy the user) into the instruments database. 

The "fake" payment handler will then show up in payment sheet for basic card. For example, https://marcospay.com: 

<img width="527" alt="Screenshot 2020-02-27 19 00 17" src="https://user-images.githubusercontent.com/870154/75423740-6b17ae00-5993-11ea-9293-439cd717e983.png">

**Comment by dlongley** [comment]
The payment handler polyfill we wrote long ago (no longer maintained) required websites to ask for permission to handle payments before they were permitted to store any payment instruments -- precisely because of this. We've done something similar with our credential handler polyfill (for the Credential Handler API in another space). So +1 for requiring websites to request permission first -- or some other clever mechanism for addressing this problem.

**Comment by msporny** [comment]
Example of how the payment handler polyfill addressed this issue: https://youtu.be/Yb-gWT1t1Rg?t=310

**Comment by marcoscaceres** [comment]
Yeah, something like @msporny demos, and @dlongley notes, is also what I have in mind.
